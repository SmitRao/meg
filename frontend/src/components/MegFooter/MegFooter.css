.footer {
    text-align: end;
    background-color: black;
    color: #efefef;
}

.footer-icons {
    font-size: xx-large;
    height: 2em;
    color: #efefef;
}

.icon {
    padding-left: 1em;
}
