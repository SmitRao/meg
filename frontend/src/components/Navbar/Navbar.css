.logo {
    width: 6em;
    height: 4.2em;
    margin-top: 1em;
    margin-bottom: 0.8em;
    background: #fff;
    /* margin: 16px 24px 16px 0; */
    float: left;
}

.menu {
    line-height: 6em;
    background: #fff;
    float: right;
}

.header {
    width: 100%;
    height: 5em;
    background: #fff;
}

