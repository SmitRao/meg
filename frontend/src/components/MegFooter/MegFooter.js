import React from "react";
import { Layout, Typography } from "antd";
import "./MegFooter.css";
import {
  FacebookFilled,
  InstagramFilled,
  TwitterOutlined,
  MailFilled
} from "@ant-design/icons";

const { Text } = Typography;

const { Footer } = Layout;

const copyrightYear = new Date().getFullYear();

class MegFooter extends React.Component {
  fbRedirect() {
    window.location.href = "https://www.facebook.com/myethicalgarment";
  }
  instaRedirect() {
    window.location.href = "https://www.instagram.com/myethicalgarment/";
  }
  twitterRedirect() {
    window.location.href = "https://twitter.com/garment_my";
  }
  emailMeg() {
    window.location.href = "mailto:sana@myethicalgarment.com";
  }

  render() {
    return (
      <Layout>
        <Footer className="footer">
          <div className="footer-icons">
            <FacebookFilled className="icon" onClick={this.fbRedirect} />
            <InstagramFilled className="icon" onClick={this.instaRedirect} />
            <TwitterOutlined className="icon" onClick={this.twitterRedirect} />
            <MailFilled className="icon" onClick={this.emailMeg} />
          </div>
          <Text style={{ color: "#efefef" }}>
            &#169; {copyrightYear}, My Ethical Garment Inc. All rights reserved.
          </Text>
        </Footer>
      </Layout>
    );
  }
}

export default MegFooter;
