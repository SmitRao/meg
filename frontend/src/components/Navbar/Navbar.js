import React from "react";
import { Layout, Menu} from "antd";
import "./Navbar.css";
import { ReactComponent as Logo } from "../../assets/logo.svg";
import { Link } from "react-router-dom";

const { Header } = Layout;

class Navbar extends React.Component {
    render() {
        return (
                <Header className="header">
                    <Link to="">
                    <Logo className="logo"/>
                    </Link>
                    <Menu
                        theme="light"
                        mode="horizontal"
                        defaultSelectedKeys={["1"]}
                        className="menu">

                        <Menu.Item key="1"> 
                        <Link to={""}> Home </Link>
                        </Menu.Item>
                        <Menu.Item key="2"> Our Values</Menu.Item>
                        <Menu.Item key="3"> Brands </Menu.Item>
                     </Menu>
                    </Header>
            )
    }
}

export default Navbar;