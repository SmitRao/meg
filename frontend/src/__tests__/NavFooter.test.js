import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import TestRenderer from 'react-test-renderer';
import Navbar from '../components/Navbar/Navbar';
import MegFooter from '../components/MegFooter/MegFooter';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure} from 'enzyme';
configure({adapter: new Adapter()});


describe('testing Navbar', () => {
    it('should render correctly', () => {
      const component = shallow(<Navbar/>);
      expect(component).toMatchSnapshot();
    });
  });

  
it('testing footer component', () => {
    const foot = TestRenderer.create(<MegFooter />);
    expect(foot).toMatchSnapshot();
});