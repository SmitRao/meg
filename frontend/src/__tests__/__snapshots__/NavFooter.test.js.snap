// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Navbar should render correctly 1`] = `
<Header
  className="header"
>
  <Link
    to=""
  >
    <ForwardRef(SvgLogo)
      className="logo"
    />
  </Link>
  <Menu
    className="menu"
    defaultSelectedKeys={
      Array [
        "1",
      ]
    }
    mode="horizontal"
    theme="light"
  >
    <MenuItem
      key="1"
    >
      <Link
        to=""
      >
         Home 
      </Link>
    </MenuItem>
    <MenuItem
      key="2"
    >
       Our Values
    </MenuItem>
    <MenuItem
      key="3"
    >
       Brands 
    </MenuItem>
  </Menu>
</Header>
`;

exports[`testing footer component 1`] = `
<section
  className="ant-layout"
>
  <footer
    className="ant-layout-footer footer"
  >
    <div
      className="footer-icons"
    >
      <span
        aria-label="facebook"
        className="anticon anticon-facebook icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="facebook"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-92.4 233.5h-63.9c-50.1 0-59.8 23.8-59.8 58.8v77.1h119.6l-15.6 120.7h-104V912H539.2V602.2H434.9V481.4h104.3v-89c0-103.3 63.1-159.6 155.3-159.6 44.2 0 82.1 3.3 93.2 4.8v107.9z"
          />
        </svg>
      </span>
      <span
        aria-label="instagram"
        className="anticon anticon-instagram icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="instagram"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M512 378.7c-73.4 0-133.3 59.9-133.3 133.3S438.6 645.3 512 645.3 645.3 585.4 645.3 512 585.4 378.7 512 378.7zM911.8 512c0-55.2.5-109.9-2.6-165-3.1-64-17.7-120.8-64.5-167.6-46.9-46.9-103.6-61.4-167.6-64.5-55.2-3.1-109.9-2.6-165-2.6-55.2 0-109.9-.5-165 2.6-64 3.1-120.8 17.7-167.6 64.5C132.6 226.3 118.1 283 115 347c-3.1 55.2-2.6 109.9-2.6 165s-.5 109.9 2.6 165c3.1 64 17.7 120.8 64.5 167.6 46.9 46.9 103.6 61.4 167.6 64.5 55.2 3.1 109.9 2.6 165 2.6 55.2 0 109.9.5 165-2.6 64-3.1 120.8-17.7 167.6-64.5 46.9-46.9 61.4-103.6 64.5-167.6 3.2-55.1 2.6-109.8 2.6-165zM512 717.1c-113.5 0-205.1-91.6-205.1-205.1S398.5 306.9 512 306.9 717.1 398.5 717.1 512 625.5 717.1 512 717.1zm213.5-370.7c-26.5 0-47.9-21.4-47.9-47.9s21.4-47.9 47.9-47.9 47.9 21.4 47.9 47.9a47.84 47.84 0 01-47.9 47.9z"
          />
        </svg>
      </span>
      <span
        aria-label="mail"
        className="anticon anticon-mail icon"
        role="img"
      >
        <svg
          aria-hidden="true"
          className=""
          data-icon="mail"
          fill="currentColor"
          focusable="false"
          height="1em"
          viewBox="64 64 896 896"
          width="1em"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-80.8 108.9L531.7 514.4c-7.8 6.1-18.7 6.1-26.5 0L189.6 268.9A7.2 7.2 0 01194 256h648.8a7.2 7.2 0 014.4 12.9z"
          />
        </svg>
      </span>
    </div>
    <span
      className="ant-typography"
      style={
        Object {
          "WebkitLineClamp": null,
          "color": "#efefef",
        }
      }
    >
      Â© 
      2020
      , My Ethical Garment Inc. All rights reserved.
    </span>
  </footer>
</section>
`;
